quixstreams/sinks/influxdb_v3.py

```diff
@@ -0,0 +1,153 @@
+
+        It batches the processed records in memory per topic partition
+        and flushes them to InfluxDB at the checkpoint.
+
+        The InfluxDB sink transparently handles backpressure if the DB instance
+        cannot accept more data at the moment
+        (e.g., when InfluxDB returns an HTTP 429 error with the "retry_after" header set).
+        When this happens, the sink will notify the Application to pause consuming
+        from the backpressured topic partition until the "retry_after" timeout elapses.
+
+        :param token: InfluxDB access token
+        :param host: InfluxDB host in format "https://<host>"
+        :param organization_id: InfluxDB organization_id
+        :param database: database name
+        :measurement: measurement name
+        :param fields_keys: a list of keys to be used as "fields" when writing to InfluxDB.
+            If empty, the whole record value will be used.
+            Default - empty.
+        :param tags_keys: a list of keys to be used as "tags" when writing to InfluxDB.
+            If empty, no tags will be sent.
+            Default - empty.
+        :param time_key: a key to be used as "time" when writing to InfluxDB.
+            By default, the record timestamp will be used with "ms" time precision.
+            When using a custom key, you may need to adjust the `time_precision` setting
+            to match.
+        :param include_metadata_tags: if True, includes record's key, topic,
+            and partition as tags.
+            Default - `False`.
+        :param batch_size: how many records to write to InfluxDB at once.
+            Note that it only affects the size of the writing batch, and not the number
+            of records flushed on each checkpoint.
+            Default - `1000`.
+        :param enable_gzip: if True, enables gzip compression for writes.
+            Default - `True`.
+        :param request_timeout_ms: an HTTP request timeout in milliseconds.
+            Default - `10000`.
+        :param debug: if True, print debug logs from InfluxDB client.
+            Default - `False`.
+        """
+
+        super().__init__()
+        self._client = InfluxDBClient3(
+            token=token,
+            host=host,
+            org=organization_id,
+            database=database,
+            debug=debug,
+            enable_gzip=enable_gzip,
+            timeout=request_timeout_ms,
+            write_client_options={
+                "write_options": WriteOptions(
+                    write_type=WriteType.synchronous,
+                )
+            },
+        )
+        self._measurement = measurement
+        self._fields_keys = fields_keys
+        self._tags_keys = tags_keys or []
+        self._include_metadata_tags = include_metadata_tags
+        self._time_key = time_key
+        self._write_precision = time_precision
+        self._batch_size = batch_size
+
+    def _iter_batches(self, it: Iterable[T], n: int) -> Iterable[Iterable[T]]:
```

Comment:
Seems to me that something like this should be something the `SinkBatch` knows how to do for you...I imagine this is gonna be a common pattern.

Reply:
On a related note (and I fully admit this is more of a subjective comment) if we also renamed `SinkBatch` to `SinkBatcher`, it helps make clear that it both collects messages _and_ generates batches for you as needed.

Admittedly a small thing, but I thought between the two ideas it might make the collective interaction/behavior more clear/streamlined.

Reply:
> Seems to me that something like this should be something the SinkBatch knows how to do for you (generating batches/chunking of certain size)...I imagine this is gonna be a common pattern.

Agree on this one, it can simplify the Influxdb sink overall

Reply:
> As to me, calling it just SinkBatch implied to me it was the entire batch

It is an entire batch for the given topic-partition. You can iterate over it and get all the batched messages.
So I'd keep the name the same.

Reply:
The `_iter_batches()` is just a utility method to limit the size of a single write to the destination.
E.g. if you need to write max 1000 items at once